src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -562,6 +562,52 @@ codet java_bytecode_convert_methodt::get_array_bounds_check(
 
 /*******************************************************************\
 
+Function: throw_arithmetic_exception
+
+  Inputs:
+
+ Outputs:
+
+ Purpose: Checks whether there is a division by zero 
+          and throws ArithmeticException if necessary
+
+\*******************************************************************/
+
+codet java_bytecode_convert_methodt::throw_arithmetic_exception(
+  const exprt &denominator,
+  const source_locationt &original_loc)
+{
+  symbolt exc_symbol;
+  if(!symbol_table.has_symbol("ArithmeticException"))
+  {
+    exc_symbol.is_static_lifetime=true;
+    exc_symbol.base_name="ArithmeticException";
+    exc_symbol.name="ArithmeticException";
+    exc_symbol.mode=ID_java;
+    exc_symbol.type=typet(ID_pointer, empty_typet());
+    symbol_table.add(exc_symbol);
```

Comment:
I'm now using a class stub. Soon, we should have a proper model so that we can capture the class hierarchy (for now, this wouldn't be caught by a handler for, say, RuntimeException).